mail_tracking/models/mail_message.py

```diff
@@ -25,27 +25,40 @@ class MailMessage(models.Model):
              " to filter tracking issues",
         default=False,
     )
-    is_failed_message = fields.Boolean(compute="_compute_is_failed_message")
+    is_failed_message = fields.Boolean(compute="_compute_is_failed_message", search='_search_is_failed_message')
 
     @api.model
     def get_failed_states(self):
         """The 'failed' states of the message"""
         return {'error', 'rejected', 'spam', 'bounced', 'soft-bounced'}
 
-    @api.depends('mail_tracking_needs_action', 'author_id', 'partner_ids',
+    @api.depends('mail_tracking_needs_action', 'author_id', 'notification_ids',
                  'mail_tracking_ids.state')
     def _compute_is_failed_message(self):
         """Compute 'is_failed_message' field for the active user"""
         failed_states = self.get_failed_states()
         for message in self:
             needs_action = message.mail_tracking_needs_action
             involves_me = self.env.user.partner_id in (
-                message.author_id | message.partner_ids)
+                message.author_id | message.notification_ids.mapped('res_partner_id'))
             has_failed_trackings = failed_states.intersection(
                 message.mapped("mail_tracking_ids.state"))
             message.is_failed_message = bool(
                 needs_action and involves_me and has_failed_trackings)
 
+    def _search_is_failed_message(self, operator, value):
+        """Search for messages considered failed for the active user.
+        Be notice that 'notificacion_ids' is a record that change if
+        the user mark the message as readed.
+        """
+        return [
+            ('mail_tracking_ids.state', 'in', list(self.get_failed_states())),
```

Comment:
Shouldn't you have to use the `operator` and the `value` to switch this condition betwen `in` and `not in`